src/main/java/com/sap/sgs/phosphor/fosstars/data/github/PackageManagement.java

```diff
@@ -0,0 +1,232 @@
+import static com.sap.sgs.phosphor.fosstars.model.value.Language.PYTHON;
+import static com.sap.sgs.phosphor.fosstars.model.value.Language.RUBY;
+import static com.sap.sgs.phosphor.fosstars.model.value.Language.SCALA;
+import static com.sap.sgs.phosphor.fosstars.model.value.Language.VISUALBASIC;
+import static com.sap.sgs.phosphor.fosstars.model.value.PackageManager.COMPOSER;
+import static com.sap.sgs.phosphor.fosstars.model.value.PackageManager.DOTNET;
+import static com.sap.sgs.phosphor.fosstars.model.value.PackageManager.GRADLE;
+import static com.sap.sgs.phosphor.fosstars.model.value.PackageManager.MAVEN;
+import static com.sap.sgs.phosphor.fosstars.model.value.PackageManager.NPM;
+import static com.sap.sgs.phosphor.fosstars.model.value.PackageManager.PIP;
+import static com.sap.sgs.phosphor.fosstars.model.value.PackageManager.RUBYGEMS;
+import static com.sap.sgs.phosphor.fosstars.model.value.PackageManager.YARN;
+
+import com.sap.sgs.phosphor.fosstars.model.Value;
+import com.sap.sgs.phosphor.fosstars.model.ValueSet;
+import com.sap.sgs.phosphor.fosstars.model.value.Language;
+import com.sap.sgs.phosphor.fosstars.model.value.Languages;
+import com.sap.sgs.phosphor.fosstars.model.value.PackageManager;
+import com.sap.sgs.phosphor.fosstars.model.value.PackageManagers;
+import com.sap.sgs.phosphor.fosstars.model.value.ValueHashSet;
+import java.io.IOException;
+import java.util.EnumMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+
+/**
+ * This data provider returns package managers which are used in a project.
+ */
+public class PackageManagement extends AbstractGitHubDataProvider {
+
+  /**
+   * A minimal number of characters in a config to consider it valid.
+   */
+  private static final int ACCEPTABLE_CONFIG_SIZE = 100;
+
+  /**
+   * Maps a programming language to its package manager.
+   */
+  private static final Map<Language, PackageManagers> KNOWN_PACKAGE_MANAGERS
+      = new EnumMap<>(Language.class);
+
+  static {
+    register(JAVA, MAVEN, GRADLE);
+    register(SCALA, MAVEN);
+    register(JAVASCRIPT, NPM, YARN);
+    register(C_SHARP, DOTNET);
+    register(F_SHARP, DOTNET);
+    register(VISUALBASIC, DOTNET);
+    register(PYTHON, PIP);
+    register(RUBY, RUBYGEMS);
+    register(PHP, COMPOSER);
+  }
+
+  /**
+   * Maps a package manager to a list of its possible config files.
+   *
+   * @see <a href="https://help.github.com/en/github/visualizing-repository-data-with-graphs/listing-the-packages-that-a-repository-depends-on">
+   *      Listing the packages that a repository depends on</a>
+   */
+  private static final Map<PackageManager, Predicate<String>[]> CONFIG_FILES_PATTERNS
+      = new EnumMap<>(PackageManager.class);
+
+  static {
+    register(MAVEN, "pom.xml"::equals);
```

Comment:
Are these not buildDescriptors? my understanding is package managers are something more than just pom.xmls or something. please correct me if I am wrong.